### ISSUE TITLE
Support for additional Postgres to JS type mapping

### ISSUE THREAD

**Issue by samwillis** [divyenduz/plv8ify#27]
Currently the type mapping are limited to basic numbers, string and bool, with a fallback to JSONB/JSON

https://github.com/divyenduz/plv8ify/blob/7a2d641bf41d24b96ba07dfd6a93f5af2f8ef12c/src/impl/PLV8ifyCLI.ts#L30-L34

It would be great to support mapping of further datatypes such as Date and typed arrays (Uint8Array).

I'm using a fork where I have added `Uint8Array` and mapped it to the `bytea` for Postgres:

```ts
  private _typeMap = {
    number: 'float8',
    string: 'text',
    boolean: 'boolean',
    Uint8Array: 'bytea',
  }
```

**Comment by divyenduz** [comment]
Totally makes sense, please feel free to make a PR for this (even hardcoding `Uint8Array: 'bytea'` is fine for now). At a later stage I think this would be configurable via a config file as well.

**Comment by mattiarossi** [comment]
Picking up this request as well, would a config file option be suitable?
I am in need of this as I will need to support custom types as well ...
The use case for us is supporting internal types and custom defined types as well .. hardcoding would not be suitable whereas the custom types could be generated in out CI during the schema types export phase that we already have in place to support using internal postgres record types ....

I was thinking about using a js config file with a typeMap definition that would then be merged with the static internal one:

types.js

```
typeMap = {
  number: 'float8',
  string: 'text',
  boolean: 'boolean',
  Uint8: 'bytea',
  test_type: 'test_type',
  'test_type[]': 'test_type[]',
}

```

input.ts:

```
interface test_type {
  name: string
  age: number
}

export function hello(test: test_type[]) {
  return {
    name: `Hello ${test[0].name}`,
    age: test[0].age,
  }
}
```


Custom types param:
```
plv8ify generate input.ts --types-config-file types.js
```

Would return

```
DROP FUNCTION IF EXISTS plv8ify_hello(test test_type[]);
CREATE OR REPLACE FUNCTION plv8ify_hello(test test_type[]) RETURNS JSONB AS $plv8ify$
// input.ts
function hello(test) {
  return {
    name: `Hello ${test[0].name}`,
    age: test[0].age
  };
}


return hello(test)

$plv8ify$ LANGUAGE plv8 IMMUTABLE STRICT;
```

**Comment by mattiarossi** [comment]
Another point with custom types, say I want to define a type/interface and re-use it across multiple functions:

```
import { test_type } from '../hello-custom-type/input'

export function hello(test: test_type[]): test_type {
  return {
    name: `Hello ${test[0].name}`,
    age: test[0].age,
  }
}


```

The function type returned from ts-morph for the included interface/type is in the form:

```
import("/Users/.../plv8ify/examples/hello-custom-type/input").test_type[]
```
and that is what the current code uses to try and find a match in the _typeMap map ... and naturally fails

I propose to improve the mapping logic by checking for the `<path>.<type>` pattern and returning only the right side of the match, if found, something like:

```
  private getTypeFromMap(type: string) {
    const typeLocal = type.split('.').pop()
    return this._typeMap[typeLocal ?? type]
  }

```

to be used instead of the direct map :
```
  private getFunctions() {
    return this._tsCompiler.getFunctions().map((fn) => {
      if (this.getFunctionTrigger(fn)) {
        fn.returnType = 'trigger'
      }
      return {
        ...fn,
        returnType: this.getTypeFromMap(fn.returnType),
      }
    })
  }
```

**Comment by mattiarossi** [comment]
I think the changes released in e636dbe003b75836f1c56af9c6bf2d912bd0bdee are enough to close this one
